src/components/organism/WithdrawPoolLiquidity/index.tsx

```diff
@@ -306,6 +308,74 @@ const WithdrawPoolLiquidity = () => {
     return percentA < percentB ? percentA : percentB;
   };
 
+  const formattedTokenBValue = () => {
+    let value = 0;
+
+    if (Number(selectedTokenB?.decimals) > 0) {
+      value = selectedTokenAssetValue?.tokenValue
+        ? new Decimal(selectedTokenAssetValue?.tokenValue).times(withdrawAmountPercentage / 100).toNumber()
+        : 0;
+    } else {
+      const percentValue = new Decimal(selectedTokenAssetValue?.tokenValue || 0)
+        .times(withdrawAmountPercentage / 100)
+        .toNumber();
+      value = percentValue < 1 ? Math.ceil(percentValue) : Math.floor(percentValue);
+    }
+
+    return value.toString();
+  };
+
+  const calculatePriceImpact = async () => {
+    if (api) {
+      if (selectedTokenNativeValue?.tokenValue !== "" && selectedTokenAssetValue?.tokenValue !== "") {
+        const poolSelected: any = pools?.find(
+          (pool: any) =>
+            pool?.[0]?.[1]?.interior?.X2?.[1]?.GeneralIndex?.replace(/[, ]/g, "") === selectedTokenB.assetTokenId
+        );
+        if (poolSelected && selectedTokenNativeValue?.tokenValue && selectedTokenAssetValue?.tokenValue) {
+          const poolReserve: any = await getPoolReserves(
+            api,
+            poolSelected?.[0]?.[1]?.interior?.X2?.[1]?.GeneralIndex?.replace(/[, ]/g, "")
+          );
+
+          const assetTokenReserve = formatDecimalsFromToken(
+            poolReserve?.[1]?.replace(/[, ]/g, ""),
+            selectedTokenB.decimals
+          );
+
+          const nativeTokenReserve = formatDecimalsFromToken(
+            poolReserve?.[0]?.replace(/[, ]/g, ""),
+            selectedTokenA.nativeTokenDecimals
+          );
+
+          const priceBeforeSwap = new Decimal(nativeTokenReserve).div(assetTokenReserve);
+
+          const priceOfAssetBForOneAssetA = new Decimal(assetTokenReserve).div(nativeTokenReserve);
+          setAssetBPriceOfOneAssetA(priceOfAssetBForOneAssetA.toFixed(5));
+
+          const valueA = new Decimal(selectedTokenNativeValue?.tokenValue).add(nativeTokenReserve);
```

Comment:
```suggestion
          const valueA = new Decimal(selectedTokenNativeValue?.tokenValue || 0).add(nativeTokenReserve);
```

Reply:
i don't think that his is necesary, we are already checking that it is not an empty string, it can be an empty string or a value bigger than 0